### ISSUE TITLE
Auto Queue features only work properly when addon window is highlighted by cursor

### ISSUE THREAD

**Issue by Krixal** [Nixxen/TrinketMenu#9]
This all works fabulously except for one massive problem: the Auto Queue features only properly trigger when I have the TrinketMenu window highlighted with my cursor. When the window isn't highlighted, the addon fails to check for any triggers and leaves all trinkets how they are. As soon as I move my cursor over the window, all Auto Queue functions work perfectly.

An example to illustrate the problem. I have Devilsaur Eye as the highest priority trinket for the first slot. The only trinket beneath it for the queue is Rune of the Guard Captain. I use Devilsaur Eye and the cog for that slot remains gold. I move cursor over the TrinketMenu window and the cog changes to grey. The Devilsaur Eye effect fades and it is not swapped out. I move my cursor over the TrinketMenu window and the Devilsaur Eye is immediately swapped for Rune of the Guard Captain. Devilsaur Eye reaches 30 seconds until ready for use and is not swapped back in. I move my cursor over the TrinketMenu window and Devilsaur Eye is immediately swapped back in. This problem drastically diminishes the utility of the addon as I have to make sure I tickle the TrinketMenu window every time it's supposed to do something on its own. 

This addon also appears to have some unlisted dependency on one of my other addons; when I disable everything except TrinketMenu for testing, the addon breaks, but it works just fine under my normal addon setup.

I'd like to thank you for working on this project. This was one of my favorite addons back in Vanilla, and having a fully functional version of it in Classic would be incredible.

**Comment by Nixxen** [comment]
This seems like a multifaceted issue (or issues), so I'll try to break it down a bit piece by piece to better understand it.

**Regarding the autoqueue issue:**
I will need to test this myself for a bit and see if I can reproduce it.
I have personally not used autoqueue much - and believed it to be broken in Classic until quite recently (well, it was very broken, but found some glaring issue that was fixed).

I'll get back to you about this once I have done some tests and either find a solution or need more info.

**Regarding dependencies:**
This sounds strange. I've had another player talk about something similar actually, but never heard back from him after requesting some troubleshooting.
It could be a "leaking" global variable that interferes with the exact same variable that another addon uses I guess, but I don't see how that could break anything if TM is the only active addon.
When you say that it breaks, does completely disable the addon in the addon selection screen on your login screen, or does it just stop working / showing up in game?
If it's the trinket menu UI "falling apart" and not displaying anything correctly any more, then this sounds like the same issue as the other person had.

Could you do the following please:
1. Make a backup of your savedvariables files for this addon.
They are located in two files - both called "TrinketMenu.lua" - in your accounts and characters "SavedVariables" folder.
These folders are located in
`<wow install path>\_classic_\WTF\Account\<accountname>\SavedVariables\`
and
`<wow install path>\_classic_\WTF\Account\<accountname>\<server>\<character>\SavedVariables\`

2. After making a backup of these two files, type this command in game.
`/trinketmenu reset`

3. If this solves the issue, please send me a copy of the backed up version of your savedvariables for TrinketMenu, as well as the newly reset versions so that I can compare them.


And finally, your thanks means a lot. Appreciate it :)

**Comment by Krixal** [comment]
> When you say that it breaks, does completely disable the addon in the addon selection screen on your login screen, or does it just stop working / showing up in game?

It does not completely disable the addon. The window for the addon does not appear under any circumstances, but the minimap cog DOES show up. However, the options page for the addon is mostly blank, and none of the addon features function properly. I can make queue lists for both trinket slots, but the addon won't actually enforce them. 

![broken options](https://user-images.githubusercontent.com/44178066/82818091-0b765500-9e53-11ea-8f6e-d95a9ebeab8d.PNG)


I made backups of the files you mentioned, and used /trinketmenu reset in game. This did not fix the issue; the addon remains nonfunctional when it is the only one loaded.

**Comment by GrayedFox** [comment]
Hi there, I've found a workaround for this issue - if you enable the "Notify when ready" checkbox you do not have to hover over the trinket in order to get the autoqueue switch to trigger.

**Comment by Krixal** [comment]
Oh hey, you're right. Thank you for pointing this out.

**Comment by GrayedFox** [comment]
@Nixxen this may provide a hint about the source of the issue... not super familiar with Lua but I thought maybe something in this function? https://github.com/Nixxen/TrinketMenu/blob/3fbb97e17eaa7b8e60b4925d50113f99130bce94/TrinketMenu.lua#L1067

**Comment by Nixxen** [comment]
Yep, I've found some strange code in this addon earlier, and it wouldn't surprise me if it is some nested check that should be somewhere else.

This definitely helps narrow down the search.
